libs/shared/ui/src/lib/components/chart/use-zoomable-chart.spec.ts

```diff
@@ -0,0 +1,615 @@
+import { act, renderHook } from '@testing-library/react'
+import { useZoomableChart } from './use-zoomable-chart'
+
+describe('useZoomableChart', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe('initial state', () => {
+    it('initializes with default zoom state', () => {
+      const { result } = renderHook(() => useZoomableChart())
+
+      expect(result.current.zoomState).toEqual({
+        left: 'dataMin',
+        right: 'dataMax',
+        refAreaLeft: '',
+        refAreaRight: '',
+      })
+      expect(result.current.zoomHistory).toEqual([])
+      expect(result.current.isCtrlPressed).toBe(false)
+      expect(result.current.isZoomed).toBe(false)
+    })
+  })
+
+  describe('keyboard event handling', () => {
+    it('sets isCtrlPressed to true when Control key is pressed', () => {
+      const { result } = renderHook(() => useZoomableChart())
+
+      act(() => {
+        const event = new KeyboardEvent('keydown', { key: 'Control' })
```

Comment:
Instead of using `KeyboardEvent`, could you use `useEvent` ? It should reproduce the user behavior. It help you to don't use each time an `act` as well